### PR TITLE
[MDS-5809] IRT table updates

### PR DESCRIPTION
**Pull request by matbusby-fw**
## Objective 

- Added support for versioning of the IRT table
- Set previous records to version 1
- Inserted new records as version 2
- Refactored frontend components to functional typescript components
- Adjusted frontend components to use the correct version based on the requirements present (preserve historical associations)

[MDS-5809](https://bcmines.atlassian.net/browse/MDS-5809)

image
image

### CODE REVIEW

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import { MineDocument } from "@mds/common/models/documents/document";
+
+interface IRTFileImportProps {
+  projectGuid: string;
+  informationRequirementsTableDocumentTypesHash: any;
+  importIsSuccessful: (success: boolean, err: string) => void;
+  downloadIRTTemplate: (url: string) => void;
+}
+
+export const IRTFileImport: FC<IRTFileImportProps & InjectedFormProps<any>> = ({
```
Comment (matbusby-fw) [inline services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx]: All changes here are refactoring from class based component, implementing typescript and refactoring to use react-redux hooks.  All functionality is the same.

**Review by henryoforeh-dev — DISMISSED** [review]

File: services/minespace-web/src/tests/components/project/informationRequirementsTablePage/__snapshots__/InformationRequirementsTablePage.spec.tsx.snap [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      "@@observable": [Function],
+      "dispatch": [Function],
+      "getState": [Function],
+      "replaceReducer": [Function],
+      "subscribe": [Function],
+    }
+  }
+>
+  <InformationRequirementsTablePage />
+</Provider>
```
Comment (simensma-fresh) [inline services/minespace-web/src/tests/components/project/informationRequirementsTablePage/__snapshots__/InformationRequirementsTablePage.spec.tsx.snap:16]: I'm wondering if we could do a full render of this to get a bit more usable snapshot?

File: services/minespace-web/src/tests/components/project/informationRequirementsTablePage/__snapshots__/InformationRequirementsTablePage.spec.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    Object {
+      "@@observable": [Function],
+      "dispatch": [Function],
+      "getState": [Function],
+      "replaceReducer": [Function],
+      "subscribe": [Function],
+    }
+  }
+>
+  <InformationRequirementsTablePage />
```
Comment (simensma-fresh) [inline services/minespace-web/src/tests/components/project/informationRequirementsTablePage/__snapshots__/InformationRequirementsTablePage.spec.tsx.snap]: I'm wondering if we could do a full render of this to get a bit more usable snapshot?

**Review by simensma-fresh — CHANGES_REQUESTED** [review]
I'm wondering if we could do a full render of this to get a bit more usable snapshot?

File: services/minespace-web/src/tests/components/project/informationRequirementsTablePage/__snapshots__/InformationRequirementsTablePage.spec.tsx.snap (hunk 3 above)
Comment (taraepp) [reply services/minespace-web/src/tests/components/project/informationRequirementsTablePage/__snapshots__/InformationRequirementsTablePage.spec.tsx.snap]: That, and probably put the [AUTHENTICATION] into the state to give the test the proper role access (I think I saw an authorization wrapper).

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx (hunk 1 above)
Comment (taraepp) [reply services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx]: My heart wants to get rid of old antd form too, but this is a great step forward.

**Review by henryoforeh-dev — DISMISSED** [review]

File: services/minespace-web/src/tests/components/project/informationRequirementsTablePage/__snapshots__/InformationRequirementsTablePage.spec.tsx.snap (hunk 3 above)
Comment (matbusby-fw) [reply services/minespace-web/src/tests/components/project/informationRequirementsTablePage/__snapshots__/InformationRequirementsTablePage.spec.tsx.snap]: Took more work than I expected to update that test, but it's looking much better now.

**Review by henryoforeh-dev — APPROVED** [review]

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx (hunk 1 above)
Comment (matbusby-fw) [reply services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx]: Switched it over!

**Review by taraepp — APPROVED** [review]
Looks good!
If it helps, for the future, instead of dispatching actions in snap test I usually just look at the redux form state (or the diff from which actions have dispatched) in dev tools and populate it with what I want to simulate conditions. I find it easier, but maybe dispatching actions makes more sense (dispatching actions may be a little more accurate).

**Review by simensma-fresh — APPROVED** [review]
🍏 
Sweet. Hopefully we can get an in-app editor in place at some point as that sql file is slightly scary 😅

**Pull request merged**